### ISSUE TITLE
[UBS Admin Cabinet ] Order work log (history)

### ISSUE THREAD

**Issue by YuliyaTymkiv** [ita-social-projects/GreenCity#3217]
**As an** administrator, **I want** to have order work log recorded **so that** I can check the order processing history

**Assumptions & Constraints**
n/a

**Preconditions**
1. User is logged in as administrator.
2. Administrator has opened an separate order from current orders view table

**Business rules**
n/a

### Acceptance Criteria

1. Order work log title is **“Історія замовлення”** and it contains the following field and their attributes:

<html>
<body>

Field Title | Field Type | Requirements
-- | -- | --
Дата | Date | DD/MM/YYYY. Date of the change event. The system automatically picks the current date
Час | Time | HH:MM. Time of the change event. Time is based on a 24 hour system, local time zone. Seconds are not given. E.g.:12:36. The system automatically picks the current time
Подія | Text | Change event. All types of characters. (described below)
Автор | Text | Author of change: **Current administrator’s Name and Surname -** if changes are made by administrator. **Система** - if changes are made automatically by the system. **Клієнт** - if changes are made by client from the order form.

</div></b>
</body>
</html>

2. All fields are mandatory and disabled. 
3. The change events are arranged in chronological order: older on the top.
4. The administrator can expand and collapse the order worklog.

![worklog collapsed ](https://user-images.githubusercontent.com/82950010/132959822-b4504d8c-8b67-4523-b544-588b63fd35bf.png)

5. First 6 change events are displayed. To see other change events the administrator should scroll using vertical scrollbar.

![worklog expend](https://user-images.githubusercontent.com/82950010/134710329-2214dc47-21b0-42b6-88e2-da70f44c9e43.png)

6. When the data in the string is more than Event column width, the end of the string should be truncated and 3 dots at the end of the string should appear.

7. By clicking on the truncated string, a popup window with the full data is shown.

8. When the admin clicks outside this popup, it should be closed.

9. Change events title and requirements:
<html>
<body>

Change event title | Requirements
-- | --
Статус замовлення - [order status] | When the order status is changed (described in user story #2901 ).    **E.g.** Статус замовлення - Сформовано; Статус замовлення - Привезе сам.    The author of Статус замовлення - Сформовано is client. **Статус замовлення - Скасовано** should be linked to the pop-up with the reason of the current order cancellation (user story #3192 ). **Статус замовлення - Не вивезено** should be linked to the pop-up with the reason of not picked up  (user story #3191 ) 
Замовлення [payment status] | When the order payment status is changed (described in user story #2897). E.g. Замовлення оплачено.
Додано коментар | When the field “Коментар”  was changed
Змінено дані Відправника | When changes were made in any of the following fields of section “Відправник”: - Ім’я; -Прізвище; -Телефон; -Email. When two or more fields were changed at the same time, it is needed to record the change event only once.
Додано порушення | When the order violation was added #2861 
Змінено деталі порушення | When the order violation details were changed #2863
Видалено порушення | When the order violation was deleted #2864
Змінено адресу вивезення відходів | When any of the following fields in section “Адреса вивезення відходів” is changed: -Вулиця; -Будинок; -Корпус; -Під’їзд; -Район. When two or more fields were changed at the same time, it is needed to record the change event only once.
Змінено деталі замовлення. [service name] [package size]: [previous quantity of packages]шт. на [current quantity of packages] шт. | When the quantity of packages was changed (story #2865). If package quantity was changed for several services, the details should be displayed  in one line and separated by ‘.’ . E.g. Змінено деталі замовлення. Безпечні відходи 120 л: 2 шт на 3 шт. Текстильні відходи 120 л: 1 шт на 0 шт.
Номер замовлення з магазину змінено з [previous eco-store order ID] на [new eco-store order ID] | When the order ID from Eco-store is changed. In case several order IDs from Eco-store were changed, the details should be displayed in one line and  separated by ‘;’.  E.g., Номер замовлення з магазину змінено з 1234567890 на 9876543210; з 8974568962 на 4658931267
Додано оплату  №[payment ID] | A new payment record was added (story #2898 acceptance criteria 6, 7): **a)** when payment confirmation was received from payment system (Fondy, LiqPay).   **b)** when payment was added manually (story #2996).  E.g. Додано оплату №4567896
Змінено деталі оплати №[payment ID] | When a payment, added manually, was changed (story #3001)  E.g. Змінено деталі оплати №4567896
Видалено оплату №[payment ID] | When a payment, added manually, was deleted (story #3000).  E.g. Видалено оплату №4567896
Використані бонуси повернено на бонусний рахунок клієнта | When the bonuses, used to pay for the order, were returned to the client’s bonus account (user story #2900, scenario 2 and 3, acceptance criteria - 3.)
Повернено кошти клієнту | When the record with comment “Повернення коштів” was added to the table with order payment details (story #2900, scenario 2, acceptance criteria 2)
Зараховано кошти на бонусний рахунок клієнта | When the record with comment “Зарахування на бонусний рахунок” was added to the table with order payment details (story #2900, scenario 1 and 3, acceptance criteria 2)
Встановлено деталі вивезення | When any field of the following fields in Section “Деталі вивезення” was set:  -Дата вивезення;  -Час вивезення;  -Станція приймання. When two or more fields were set at the same time, it is needed to record the change event only once.
Змінено деталі вивезення.[Дата вивезення:.....Час вивезення:.....Станція приймання:.........] | When any field of the following fields in Section “Деталі вивезення” was changed:  -Дата вивезення;  -Час вивезення;  -Станція приймання. When two or more fields were changed at the same time, it is needed to record the change event only once (e.g. Змінено деталі вивезення. Дата вивезення: 07.03.2023 Час вивезення: 08-00:09-00 Станція приймання: Саперно-Слобідська)
Закріплено менеджера обдзвону | When a call-center manager  was assigned
Змінено менеджера обдзвону | When assigned call-center manager was changed to another one
Закріплено логіста | When a logist  was assigned
Змінено логіста | When assigned logist was changed to another one
Закріплено штурмана | When a call-center pilot  was assigned
Змінено штурмана | When assigned pilot was changed to another one
Закріплено водія | When a driver was assigned
Змінено водія | When assigned driver was changed to another one

</div></b>
</body>
</html>

Mockup:
https://www.figma.com/file/092AhCGadc1Hq8nAysxmYB/ITA-Greencity-УБС?node-id=17741%3A254

**Out of scope**

**Epic link**
E.g.: Epic #2195 

**Labels to be added**
"User story", Correspondind Functionality (e.g. User), Priority ("pri: "), "Corresponding Epic"

### Tasks 
1. - [x] sample task.
